File: 1086-divisor-game/divisor-game.py
class Solution:
    def recur(self, n: int, alice: bool) -> bool:
        if n == 0:
            return 
        for i in range(1, n):
            if n%i == 0:
                alice[0] = not alice[0]
                self.recur(n-i, alice)
                break

    def divisorGame(self, n: int) -> bool:
        alice = [True]
        self.recur(n, alice)
        return not alice[0]

File: 1086-divisor-game/divisor-game.c
void recur(int n, bool *alice){
    if (n == 0){
        return ;
    }
    for (int i=1; i<n; i+=1){
        if (n%i == 0){
            *alice = !(*alice);
            recur(n-i, alice);
            break;
        }
    }
}
bool divisorGame(int n) {
    bool alice = true;
    recur(n, &alice);
    return !alice;
}